crates/biome_service/src/diagnostics.rs

```diff
@@ -56,6 +57,10 @@ pub enum WorkspaceError {
     Vcs(VcsDiagnostic),
     /// Diagnostic raised when a file is protected
     ProtectedFile(ProtectedFile),
+    /// An invalid patterns was given
+    ParsePatternError(ParseError),
+    /// No pattern with the given ID
+    InvalidPattern(InvalidPattern),
```

Comment:
A suggestion: have a `SearchError` instead and then an `enum SearchError` that will contain these two variants. 

The search expresses the error better because that's what the feature is about. The "pattern" thing is a detail.